src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, ChangeEventHandler, ChangeEvent } from "react";
+import { db as firebaseDB } from "../services/firebase-client";
+import { sha256 } from "js-sha256";
+
+export const RegistrationForm = ({ onRegistrationDone = (() => {}) }) => {
+  
+  const [username, setUsername] = useState("");
+  const [redemptionType, setRedemptionType] = useState("");
+
+  const InputChangeHandlers: { [key: string]: ChangeEventHandler<HTMLInputElement> } = {
+    username: (e: ChangeEvent<HTMLInputElement>) => {
+      setUsername(e.target.value);
+    },
+    redemptionType: (e: ChangeEvent<HTMLInputElement>) => {
+      setRedemptionType(e.target.value);
+    },
+  };
+  const createInputChangeHandler = ( name: string ): ChangeEventHandler<HTMLInputElement> => {
+    const handleInputChange = InputChangeHandlers[name] || ((event: ChangeEvent<HTMLInputElement>) => {});
+
+    return handleInputChange;
+  };
+  const registerUser = async () => {
+    const createDate = new Date();
+    const redemptionString = `${username}_${redemptionType}-${createDate}`;
+    const redemptionCode = sha256(redemptionString)
+
+    try {
+      await firebaseDB.append("users", {
+        username,
+        redemptionType,
+        redemptionCode,
+        createDate,
+      });
+
+      onRegistrationDone();
+    } catch (e) {
+      console.error(e)
+    }
+  };
+
+  return (
+    <form>
+      <h2>Registration Form</h2>
+      <input
+        type="text"
+        name="username"
+        value={username}
+        onChange={createInputChangeHandler("username")}
```

Comment:
`create` is an action that is going to consume resources.

Also, the convention is to use `handle` as prefix of the name of an event handler

https://stackoverflow.com/questions/60048249/what-is-the-right-name-of-event-handler-onclick-or-handleclick
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364